optional/optional.py

```diff
@@ -1,12 +1,40 @@
-from .nothing import Nothing
-from .something import Something
+from typing import TypeAlias, TypeVar, overload
 
+from optional.nothing import Nothing
+from optional.something import Something
 
-class Optional(object):
-    @classmethod
-    def of(cls, thing=None):
-        return Nothing(cls) if thing is None else Something(thing, cls)
+T = TypeVar("T")
 
-    @classmethod
-    def empty(cls):
-        return Nothing(cls)
+
+Option: TypeAlias = Something[T] | Nothing
+
+
+class Optional:
+    """Entrypoint of optional.py
+
+    Provides static methods to help intelligently create
+    both :class:`Something` and :class:`Nothing` values"""
+
+    @staticmethod
+    @overload
```

Comment:
I assume these overloads somehow offload to the single implementation?

Reply:
Yeah. It's for the typing. Consider that `of` technically takes a `(T | None)` and returns an `Option[T]`, but that type signature doesn't make it clear that `None -> Nothing` and `T -> Option[T]`. These overloads encode that information